### PR TITLE
Hardware Key Support docs - additional troubleshooting info

### PR DESCRIPTION
**Pull request by Joerger**
Update hardware key support docs to cover re-login flow from role setting and add additional troubleshooting info.

This update is based on @stevenGravy's experience trying Hardware Key Support.

Updates https://github.com/gravitational/teleport/issues/20384

### CODE REVIEW

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 1]
```diff
@@ -107,22 +108,22 @@ cluster auth preference has been updated
 
 Once hardware key support is enforced, affected users will be required to have their login sessions backed by a Hardware Key for all Teleport requests.
 
-These users will be prompted to connect and touch their YubiKey on log in:
+Affected users will be prompted to connect and touch their YubiKey on log in. If this is their first login, they may be required to immediately relogin.
 
 <ScopedBlock scope={["oss"]}>
```
Comment (stevenGravy) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:113]: We only support Enterprise and Cloud so this can be removed.  It won't be visible

Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:113]: Better to show it than have blanks in the page, this was an issue in the past - https://github.com/gravitational/teleport/issues/24211

**Review by lsgunn-teleport — COMMENTED** [review]
I made some wording suggestions in this PR.
https://github.com/gravitational/teleport/pull/29157
There were a couple of typos (like "have have" and "in order to perform safely authenticate you given your authentication settings"). I might also suggest using the more generic hardware security key instead of mentioning Yubikey explicitly (except where it's necessary).

**Review by lsgunn-teleport — APPROVED** [review]
LGTM 😄.

**Review by stevenGravy — APPROVED** [review]

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 2]
```diff
@@ -9,45 +9,56 @@ description: Hardware Key Support
...
+To prevent these types of attacks, Teleport supports hardware-based private keys. 
+Unlike disk-based private keys, hardware-based private keys are generated and stored directly 
+on a hardware device and are impossible to export. 
+With hardware-based private keys, a login session is only functional if there's also access to the 
+hardware device where the key was generated and stored.
 
-Additionally, this feature can be configured to require touch for every Teleport request, including non-session requests. With touch required, hardware key support provides better security in comparison to our per-session MFA offering.
+Additionally, you can configure this feature to require a touch for every Teleport request, including non-session requests. 
+With touch required, hardware key support provides better security than per-session MFA offering.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:27]:
```suggestion
With touch required, hardware key support provides better security than per-session MFA.
```

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 3]
```diff
@@ -9,45 +9,56 @@ description: Hardware Key Support
   Hardware Key Support requires Teleport Enterprise.
 </Admonition>
 
-By default, `tsh`, Teleport Connect, and other Teleport clients store a user's login session directly on their filesystem. If a user's filesystem is compromised, any of their active Teleport login sessions would also be compromised.
+By default, `tsh`, Teleport Connect, and other Teleport clients store a user's login session directly on the local filesystem.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:12]: It's not really the "session" that's stored on the filesystem, but the certificates. Should we be specific?

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 4]
```diff
@@ -9,45 +9,56 @@ description: Hardware Key Support
...
+    - Application access.
 
-  The unsupported features above will not function for users that have have hardware key support
-  enforced. This is either because the user's hardware key can't be accessed within that feature
-  (WebUI) or because the protocol only supports raw private keys currently (Agent forwarding,
-  Kubernetes access, desktop access). 
+  If you require users to have a hardware key to access your infrastructure, they won't be able to
+  use any of the unsupported features either because the hardware key can't be accessed—for example, 
+  through the Teleport Web UI—or because the protocol only supports raw private keys. 
+  For example, agent forwarding. Kubernetes access, and desktop access all currently require raw private keys.
```
[the remark this answers is not in the document]
Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:55]: I'll remove the examples as they are listed above

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 5]
```diff
@@ -97,32 +113,34 @@ spec:
...
+Affected users will be prompted to connect and touch their YubiKey to sign in. 
+The first time users sign in with their hardware key they might be required to immediately sign in again.
 
 <ScopedBlock scope={["oss"]}>
 
 ```code
 $ tsh login --user=dev --proxy=proxy.example.com:3080
 # Enter password for Teleport user dev:
+# Unmet private key policy "hardware_key_touch".
+# Re-initiating login with YubiKey generated private key.
```
[the remark this answers is not in the document]
Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:136]: I can update this in a follow up PR.

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 6]
```diff
@@ -176,22 +197,82 @@ $ tsh clusters
...
+all access requests rather than just session requests. 
+Although upgrading to `hardware_key_touch` provides the best security, you can get a similar level of security 
+by combining hardware key support with per-session MFA.
 
-This prevents basic data exfiltration attacks for Teleport session and non-session requests. However, it does not prevent hacking attacks with non-session Teleport requests, since gaining access to a user's computer would provide both their certificates on disk and a local connection to their YubiKey, assuming its connected.
+You can configure this combined functionality by specifying `require_session_mfa: hardware_key` in the `role` and `cluster_auth_preference` settings. 
+This option continues to use per-session MFA checks for starting sessions and requires hardware-based private keys for all Teleport requests.
 
+This configuration prevents basic data exfiltration attacks for Teleport sessions and non-session requests. 
+However, it does not prevent attacks with non-session Teleport requests if an attacker gets access to a
```
[the remark this answers is not in the document]
Comment (Joerger) [reply docs/pages/access-controls/guides/hardware-key-support.mdx:211]: I'll referece `tctl edit` as an example.

**Comment by alexfornuto** [comment]
@Joerger looks like this is still waiting on approval from members of two no-docs groups. This would be a good candidate to request for review in #docs.

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 7]
```diff
@@ -9,45 +9,56 @@ description: Hardware Key Support
   Hardware Key Support requires Teleport Enterprise.
 </Admonition>
 
-By default, `tsh`, Teleport Connect, and other Teleport clients store a user's login session directly on their filesystem. If a user's filesystem is compromised, any of their active Teleport login sessions would also be compromised.
+By default, `tsh`, Teleport Connect, and other Teleport clients store a user's key and certificates directly
+on their filesystem. If a user's filesystem is compromised, any of their active Teleport User keys and certificates
```
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:13]: Why is "User" capitalized here?

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 8]
```diff
@@ -9,45 +9,56 @@ description: Hardware Key Support
...
+would also be compromised.
 
-[Per-session MFA](per-session-mfa.mdx) provides a way to require an MFA presence check with a user's login session, but this check is only used for starting sessions with Teleport services (SSH, Kube, DB, etc.). This means that non-session actions, like administrative commands run with `tctl`, can still be run with an exfiltrated login session. 
+You can configure [per-session MFA](per-session-mfa.mdx) to require a second factor authentication check 
+when users start new sessions with Teleport services, such as SSH, Kubernetes, DB, and so on.
```
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:17]:
```suggestion
when users start new sessions with Teleport services, such as the SSH Service, Kubernetes Service, Database Service, and so on. 
```
So we have consistent architectural terms throughout the docs

**Review by ptgott — APPROVED** [review]

**Review by AntonAM — APPROVED** [review]

**Review by GavinFrazar — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**